document-store/src/main/java/org/hypertrace/core/documentstore/expression/impl/KeyExpression.java

```diff
@@ -0,0 +1,33 @@
+package org.hypertrace.core.documentstore.expression.impl;
+
+import static org.hypertrace.core.documentstore.commons.DocStoreConstants.IMPLICIT_ID;
+
+import com.google.common.base.Preconditions;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Value;
+import org.apache.commons.lang3.StringUtils;
+import org.hypertrace.core.documentstore.Key;
+import org.hypertrace.core.documentstore.expression.type.FilterTypeExpression;
+import org.hypertrace.core.documentstore.parser.FilterTypeExpressionVisitor;
+
+@Value
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class KeyExpression implements FilterTypeExpression {
+  Key key;
+
+  public static KeyExpression of(final Key key) {
+    Preconditions.checkArgument(key != null, "key is null");
+    return new KeyExpression(key);
+  }
+
+  @Override
+  public <T> T accept(final FilterTypeExpressionVisitor visitor) {
+    return visitor.visit(this);
+  }
+
+  @Override
+  public String toString() {
+    return IMPLICIT_ID + " = " + StringUtils.wrap(key.toString(), "'");
```

Comment:
Do we use `StringUtils.wrap` to avoid injection attacks? In other words, can `key` be maliciously crafted, since we are doing a basic string concatenation and not using string builders, or is it taken care of in the pipeline later?

Or, is `toString` method being used mostly for logging purposes?

Reply:
>  is toString method being used mostly for logging purposes?

That's correct. `toString()` is only for logging (except the FunctionalExpression.toString() in the case of MongoDB, which is escaped during the parsing step). We are using PreparedStatement in `PostgresFilterTypeExpressionVisitor.visit(KeyExpression)` during parsing.